File: src/main/java/com/exsoloscript/mysteryteams/module/BannerModule.java
package com.exsoloscript.mysteryteams.module;

import com.exsoloscript.mysteryteams.team.MysteryTeam;
import org.bukkit.Material;
import org.bukkit.block.Banner;
import org.bukkit.event.EventHandler;
import org.bukkit.event.inventory.PrepareItemCraftEvent;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.BannerMeta;

public class BannerModule extends TeamBasedModule {

    public BannerModule() {
        super("Banners");
    }

    @Override
    public ItemStack getItem(MysteryTeam t) {
        ItemStack i = new ItemStack(Material.BANNER);

        BannerMeta bm = (BannerMeta) i.getItemMeta();
        bm.setBaseColor(t.getColorData().getDyeColor());
        i.setItemMeta(bm);

        return i;
    }

    @EventHandler
    public void onPrepareCraft(PrepareItemCraftEvent event) {
        if (event.getInventory().getResult().getType() == Material.BANNER)
            event.getInventory().getResult().setType(Material.AIR);
    }
}

File: src/main/kotlin/com/exsoloscript/mysteryteams/module/BannerModule.kt
package com.exsoloscript.mysteryteams.module

import com.exsoloscript.mysteryteams.team.MysteryTeam
import org.bukkit.Material
import org.bukkit.event.EventHandler
import org.bukkit.event.inventory.PrepareItemCraftEvent
import org.bukkit.inventory.ItemStack
import org.bukkit.inventory.meta.BannerMeta

class BannerModule : TeamBasedModule("Banners") {

    override fun getItem(t: MysteryTeam): ItemStack {
        val i = ItemStack(Material.BANNER)

        val bm = i.itemMeta as BannerMeta
        bm.baseColor = t.colorData.dyeColor
        i.itemMeta = bm

        return i
    }

    @EventHandler
    fun onPrepareCraft(event: PrepareItemCraftEvent) {
        if (event.inventory.result != null) {
            if (event.inventory.result.type == Material.BANNER)
                event.inventory.result = ItemStack(Material.AIR)
        }
    }
}
